### ISSUE TITLE
Generate graph pb file from model 

### ISSUE THREAD

**Issue by dan-l** [napon/tensorflow-test#1]
This should generate `graph.pb`.

```
import keras.backend.tensorflow_backend as K
from tensorflow.python.framework import graph_util

sess = tf.Session()
K.set_session(sess)

model = build_model()
PRETRAINED_MODEL_PATH = 'salamon-cnn-model.h5'
model.load_weights(PRETRAINED_MODEL_PATH)

# We retrieve the protobuf graph definition
graph = tf.get_default_graph()
input_graph_def = graph.as_graph_def()
output_node_names = "Predictions"

# We use a built-in TF helper to export variables to constants
output_graph_def = graph_util.convert_variables_to_constants(
   sess, # The session is used to retrieve the weights
   input_graph_def, # The graph_def is used to retrieve the nodes,
   [n.name for n in input_graph_def.node]
)

# Finally we serialize and dump the output graph to the filesystem
with tf.gfile.GFile('graph.pb', "wb") as f:
    f.write(output_graph_def.SerializeToString())
print("%d ops in the final graph." % len(output_graph_def.node))
```

**Comment by realwol** [comment]
For the code above, should I take that the h5 file contains just the dataset, the build_model def load the model, Am I right?

**Comment by napon** [comment]
Sorry for the late response @realwol, its been a while since I touched this code but I believe the .h5 file contains pre-determined weights for each feature. The dataset is not included here.

CC @dan-l

**Comment by yesuani** [comment]
Hello napon
how Generate graph pb file from model.
from this image classification class:

from keras.applications.resnet50 import ResNet50
from keras.preprocessing import image
from keras.applications.resnet50 import preprocess_input, decode_predictions
import numpy as np

model = ResNet50(weights='imagenet')

img_path = 'cesariaevora.jpg'
img = image.load_img(img_path, target_size=(224, 224))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

preds = model.predict(x)
# decode the results into a list of tuples (class, description, probability)
# (one such list for each sample in the batch)
print('Predicted:', decode_predictions(preds, top=3)[0])
